### PR TITLE
Feat(eos_cli_config_gen): Refactor STP template and support STP root primary/secondary

### LINKED ISSUE #1329: Support for stp root primary/secondary commands
### Enhancement summary

Asking to add support for the following STP CLI commands:

```
spanning-tree root primary/secondary
spanning-tree mst <0-4094> root primary/seconday
```
Today it is possible to set the priority manually to a value but it would be useful to add support also for these commands:
https://avd.sh/en/releases-v3.x.x/roles/eos_cli_config_gen/index.html#spanning-tree

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

I would like to set these commands for an active-standby scenario and align the active PE with the primary STP root without the need of setting the priority explicitly.

### Describe the solution you would like

Something along these lines:
```
spanning_tree:
  root: < primary | secondary >
  mst:
     root: < primary | secondary >
```

### Describe alternatives you have considered

Setting the priority value is an alternative and it is supported.

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by tgodaA**
## Change Summary

Support STP root primary or secondary commands and refactor stp template according to cli.

## Related Issue(s)

Fixes #1329 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Refactor template output according to cli

Old output:

```
spanning-tree root super
spanning-tree mode mstp
no spanning-tree vlan-id 4094
```

New output:

```
spanning-tree mode mstp
no spanning-tree vlan-id 4094
spanning-tree root super
```

Added new knobs for spanning tree to support root primary/secondary:

Old model:

```
spanning_tree:
  mst_instances:
    # Use '-' to specify range of instance ids, like 200-210
    "< instance_id(s) >":
      priority: < priority >
```

```
spanning_tree:
  root: < primary | secondary >
  mst_instances:
    # Use '-' to specify range of instance ids, like 200-210
    "< instance_id(s) >":
      priority: < priority >
      root: < primary | secondary >
```

## How to test
Molecule

## Checklist

- [x] Update artifacts for the refactored template
- [x] Verify the STP output configuration over cli
- [x] Update documentation  

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/spanning-tree.j2 [hunk 1]
```diff
@@ -10,22 +10,40 @@ STP mode: **{{ spanning_tree.mode | arista.avd.default("mstp") }}**
 
 STP Root Super: **{{ spanning_tree.root_super }}**
 {%     endif %}
+{%     if spanning_tree.root is arista.avd.defined %}
+
+STP Root: **{{ spanning_tree.root }}**
+{%     endif %}
 {%     if spanning_tree.mst_instances is arista.avd.defined %}
 
 ### MSTP Instance and Priority
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/spanning-tree.j2:19]: I would suggest changing this heading to `MSTP Instances` and then combine the two tables below, so it shows priority if set, and check for root primary/secondary if priority is not set. (Like done in config template)

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/spanning-tree.j2:19]: Maybe add the instance `vlans` to the table while we are at it. (move from below in another table)

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/spanning-tree.j2:19]: Done.

**Review by ClausHolbechArista — COMMENTED** [review]
I don't think we should support these knobs, since they are actually aliases to the priority commands in EOS:
```cli
R3(config)#show run | sec spanning-tree
spanning-tree mode mstp
R3(config)#spanning-tree root primary
R3(config)#show run | sec spanning-tree
spanning-tree mode mstp
spanning-tree mst 0 priority 8192
R3(config)#
```

**Comment by tgodaA** [comment]
Yeah, these are aliases to priority. But adding primary or secondary commands to EOS makes it easy for the user. In background they convert to priority. Primary - 8192, Secondary 16384. Also I didn't add the converted priority to the documentation but added only if it's primary or secondary because the default value might change anytime.

**Comment by xaviramon** [comment]
I don't like the idea of root primary or secondary. Do we have a use case or special need for it? I would prefer to set the STP priority manually, also because different manufacturers could use a different alias with a different priority.

**Comment by tgodaA** [comment]
I am waiting for suggestion from Claus. If this doesn't look good, please close it or comment to close it.

**Comment by tgodaA** [comment]
The same feat can achieved by using priority. So not adding this feature.

**Pull request closed without merging**